shared-module/audiocore/Mixer.c

```diff
@@ -152,31 +116,31 @@ uint32_t add8signed(uint32_t a, uint32_t b) {
 }
 
 uint32_t add8unsigned(uint32_t a, uint32_t b) {
-    #if (defined (__ARM_FEATURE_DSP) && (__ARM_FEATURE_DSP == 1))
+    /*#if (defined (__ARM_FEATURE_DSP) && (__ARM_FEATURE_DSP == 1))
```

Comment:
Why did you give up on the saturating arithmetic builtins? They are tiny and will help save space.

Reply:
In short: I don't currently understand it enough to accomplish it. These were originally done by DeanM, but only the 16 bit signed funcs were complete. The rest were mostly stubs/initial.

Reply:
OK, looking at this in more detail, the saturated builtins are only available on M4, not M0. There's a lot of stuff online about how to implement unsigned and signed saturated arithmetic. This is good (original web page is blank at the moment): https://webcache.googleusercontent.com/search?q=cache:bSrCZnMkEfkJ:https://locklessinc.com/articles/sat_arithmetic/+&cd=3&hl=en&ct=clnk&gl=us

There are a few example libraries in github but mostly in C++, unfortunately. The code here could maybe be adapted for 8 and 16 bits: https://github.com/esote/bsa/blob/master/bsa.c
https://stackoverflow.com/questions/121240/how-to-do-unsigned-saturating-addition-in-c
https://stackoverflow.com/questions/17580118/signed-saturated-add-of-64-bit-ints
etc.

Reply:
i srsly doubt this would fit in m0 anyways, lets have it for m4 only!

Reply:
if it's only for m4, you can just use the builtins and not bother to figure out other implementations. Just throw a compile `#error` for now or something like that.

Reply:
The only M0 builds that include this are express boards with `audiocore` turned on (via `audioio`). 

As of now, there are 5 build failures; all 5 only fail the Pinyin translated versions. I used a Feather M0 Express during dev/testing. The en_US build currently has `4168 bytes free in flash` (in my env).

If not including `MixerVoice` on M0 is really how we want to proceed, it will require maintaining the current _Mixer only_ code for M0, and the _Mixer+MixerVoice_ for the M4 (using `#ifdefs`). Otherwise, it'll all need to be moved out of `audiocore` to gain a more granular control.

Reply:
I'm ok either turning of underused modules like bitbangio or i2cslave to make this fit. It'd also be ok to move it to `audiomixer` module to give ourselves flexibility. (I want to do `audiosynth` at some point too.)

Reply:
New state of things accounting for all of this, plus the move to `audiomixer`:

- `audiomixer` is turned off completely for M0. This is a breaking change to `audioio.Mixer`.
- ARMv7M intrinsic functions are used where available (all addition functions, and 16bit signed multiplication). Also changed the pre-processor flags; the intrinsics used are in CMSIS-M, not CMSIS-DSP.
- C functions are used where intrinsic isn't available. They are also available as alternates to supported ARMv7M intrinsics, should they be desired in the future. Dan pointed me to some gcc builtins that may help slim them down (https://gcc.gnu.org/onlinedocs/gcc/Integer-Overflow-Builtins.html), which I plan on testing.
- Made better `TODO`s, I hope.